l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -0,0 +1,328 @@
+                IL1Bridge(baseTokenBridge[_request.chainId]).bridgehubDepositBaseToken{value: _request.mintValue}(
+                    _request.chainId,
+                    msg.sender,
+                    token,
+                    _request.mintValue
+                );
+            } else {
+                require(msg.value == 0, "Bridgehub: non-eth bridge with msg.value");
+                // note we have to pass token, as a bridge might have multiple tokens.
+                IL1Bridge(baseTokenBridge[_request.chainId]).bridgehubDepositBaseToken(
+                    _request.chainId,
+                    msg.sender,
+                    token,
+                    _request.mintValue
+                );
+            }
+        }
+
+        address stateTransition = getZkSyncStateTransition(_request.chainId);
+        canonicalTxHash = IZkSyncStateTransition(stateTransition).bridgehubRequestL2Transaction(
+            msg.sender,
+            _request.l2Contract,
+            _request.mintValue,
+            _request.l2Value,
+            _request.l2Calldata,
+            _request.l2GasLimit,
+            _request.l2GasPerPubdataByteLimit,
+            _request.factoryDeps,
+            _request.refundRecipient
+        );
+    }
+
+    /// @notice After depositing funds to the baseTokenBridge, the secondBridge is called
+    ///  to return the actual L2 message which is sent to the Mailbox.
+    ///  This assumes that either ether is the base token or
+    ///  the msg.sender has approved the baseTokenBridge with the mintValue,
+    ///  and also the necessary approvals are given for the second bridge.
+    /// @notice The logic of this bridge is to allow easy depositing for bridges.
+    /// Each contract that handles the users ERC20 tokens needs approvals from the user, this contract allows
+    /// the user to approve for each token only its respective bridge
+    /// @notice This function is great for contract calls to L2, the secondBridge can be any contract.
+    function requestL2TransactionTwoBridges(
+        L2TransactionRequestTwoBridgesOuter calldata _request
+    ) public payable override nonReentrant returns (bytes32 canonicalTxHash) {
+        {
+            address token = baseToken[_request.chainId];
+
+            if (token == ETH_TOKEN_ADDRESS) {
+                require(msg.value == _request.mintValue + _request.secondBridgeValue, "Bridgehub: msg.value mismatch");
+                // kl todo it would be nice here to be able to deposit weth instead of eth
+                IL1Bridge(baseTokenBridge[_request.chainId]).bridgehubDepositBaseToken{value: _request.mintValue}(
+                    _request.chainId,
+                    msg.sender,
+                    token,
+                    _request.mintValue
+                );
+            } else {
+                require(msg.value == _request.secondBridgeValue, "Bridgehub: msg.value mismatch 2");
```

Comment:
Does second bridge have to accept ETH? Maybe it accepts ERC20s

Reply:
`secondBridgeValue` is just a `msg.value` that needs to be passed to the second bridge